iac-common/src/test/java/org/sonar/iac/common/yaml/YamlSensorTest.java

```diff
@@ -127,13 +133,40 @@ void toParseException_NullInputFile() {
     Exception exception = mock(Exception.class);
     when(exception.getMessage()).thenReturn("message");
 
-    ParseException e = sensor().toParseException("action", null, exception);
+    ParseException e = sensor().toParseException("action", new InputFileContext(null, null), exception);
     assertThat(e)
       .hasMessage("Cannot action 'null'")
       .extracting(ParseException::getPosition)
       .isNull();
   }
 
+  @Test
+  void shouldThrowExceptionWhenFailFastIsTrue() {
+    MarkedYamlEngineException yamlEngineException = mock(MarkedYamlEngineException.class);
+    when(yamlEngineException.getProblemMark()).thenReturn(Optional.of(new Mark("mark", 1, 1, 1, new int[0], 1)));
+    when(yamlEngineException.getMessage()).thenReturn("message");
+
+    InputFile inputFile = mock(InputFile.class);
+    when(inputFile.toString()).thenReturn("TestFile");
+    when(inputFile.newPointer(2, 0)).thenReturn(new DefaultTextPointer(1, 0));
+    var sensorContext = createSensorContextFailFast();
+    var inputFileContext = new InputFileContext(sensorContext, inputFile);
+    when(inputFile.newPointer(anyInt(), anyInt())).thenThrow(IllegalArgumentException.class);
+
+    var e = catchException(() -> sensor().toParseException("action", inputFileContext, yamlEngineException));
+    assertThat(e)
+      .isInstanceOf(IllegalStateException.class)
+      .hasMessage("Unable to create new pointer for TestFile position 2:0");
+  }
+
+  private static SensorContext createSensorContextFailFast() {
+    var sensorContext = mock(SensorContext.class);
+    Configuration config = mock(Configuration.class);
+    when(config.getBoolean("sonar.internal.analysis.failFast")).thenReturn(Optional.of(true));
+    when(sensorContext.config()).thenReturn(config);
+    return sensorContext;
+  }
```

Comment:
Please use the SensorContextTester instead of a hard to debug mock solution.

```
var context = SensorContextTester.create(Paths.get("."));
    var config = new MapSettings().setProperty("sonar.internal.analysis.failFast", true);
    context.setSettings(config);
    return context;
```